driver/src/driver/scheduler.rs

```diff
@@ -66,11 +72,21 @@ impl<'a> Scheduler<'a> {
         driver: &'a mut dyn StableXDriver,
         auction_timing_configuration: AuctionTimingConfiguration,
     ) -> Self {
-        assert!(auction_timing_configuration.latest_solve_attempt_time <= BATCH_DURATION);
+        assert!(
+            auction_timing_configuration.solver_time_limit < SOLVING_WINDOW,
+            "The solver time limit must be within the solving window",
+        );
+        assert!(
+            auction_timing_configuration.latest_solve_attempt_time
+                < auction_timing_configuration.solver_time_limit,
+            "The latest solve attempt time must be earlier than the solver time limit",
+        );
         assert!(
             auction_timing_configuration.target_start_solve_time
-                < auction_timing_configuration.latest_solve_attempt_time
+                < auction_timing_configuration.latest_solve_attempt_time,
+            "the target solve start time must be earlier than the latest solve time",
```

Comment:
ubernit: last message here is the only one whose first character is not capitalized. 

If this does wind up eventually being changed, I would even go as far as removing all of the "The " from each of the messages since they only contribute width.

Reply:
> I would be in favour of using system time directly from the machine this is running on since its essentially equivalent (given that the EVM time is the miner's system time). This would eliminate a completely unnecessary call to the EVM.

I believe this PR was for calculating solver time limit using the system time and not using EVM time. As far as I can tell, there is no new EVM calls added by this PR.

> last message here is the only one whose first character is not capitalized.

Sorry, that was my bad! I tend to not capitalize log messages and errors and didn't realize that we were in the rest of the file. 

I think as a follow up, we could go over our logs and clean up the style everywhere so they are consistent.